phpBB/phpbb/profilefields/manager.php

```diff
@@ -499,4 +515,170 @@ public function build_insert_sql_array($cp_data)
+	*/
+	public function disable_profilefields($profilefield_type_name)
+	{
+		// Get the list of active profile fields of this type
+		$profile_fields = $this->list_profilefields($profilefield_type_name, true);
+
+		// If no profile fields affected, then nothing to do
+		if (!sizeof($profile_fields))
+		{
+			return;
+		}
+
+		// Update the affected profile fields to "inactive"
+		$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . '
+			SET field_active = 0
+			WHERE field_active = 1
+				AND ' . $this->db->sql_in_set('field_id', array_keys($profile_fields));
+		$this->db->sql_query($sql);
+
+		// Save modified information into a config_text field to recover on enable
+		$this->config_text->set($profilefield_type_name . '.saved', json_encode($profile_fields));
+
+		// Log activity
+		foreach ($profile_fields as $field_ident)
+		{
+			add_log('admin', 'LOG_PROFILE_FIELD_DEACTIVATE', $field_ident);
+		}
+	}
+
+	/**
+	* Purge all profile fields of a certain type
+	*
+	* This should be called when an extension which has profile field types
+	* is purged so that all those profile fields are removed
+	*
+	* @param string $profilefield_type_name Type identifier of the profile fields
+	*/
+	public function purge_profilefields($profilefield_type_name)
+	{
+		// Remove the information saved on disable in a config_text field, not needed any longer
+		$this->config_text->delete($profilefield_type_name . '.saved');
+
+		// Get the list of all profile fields of this type
+		$profile_fields = $this->list_profilefields($profilefield_type_name);
+
+		// If no profile fields exist, then nothing to do
+		if (!sizeof($profile_fields))
+		{
+			return;
+		}
+
+		$this->db->sql_transaction('begin');
+
+		// Delete entries from all profile field definition tables
+		$where = $this->db->sql_in_set('field_id', array_keys($profile_fields));
+		$this->db->sql_query('DELETE FROM ' . PROFILE_FIELDS_TABLE . ' WHERE ' . $where);
+		$this->db->sql_query('DELETE FROM ' . PROFILE_FIELDS_LANG_TABLE . ' WHERE ' . $where);
+		$this->db->sql_query('DELETE FROM ' . PROFILE_LANG_TABLE . ' WHERE ' . $where);
+
+		// Drop columns from the Profile Fields data table
+		foreach ($profile_fields as $field_ident)
+		{
+			$this->db_tools->sql_column_remove(PROFILE_FIELDS_DATA_TABLE, 'pf_' . $field_ident);
+		}
+
+		// Reset the order of the remaining fields
+		$order = 0;
+
+		$sql = 'SELECT *
+			FROM ' . PROFILE_FIELDS_TABLE . '
+			ORDER BY field_order';
+		$result = $this->db->sql_query($sql);
+
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$order++;
+			if ($row['field_order'] != $order)
+			{
+				$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . "
+					SET field_order = $order
+					WHERE field_id = {$row['field_id']}";
+				$this->db->sql_query($sql);
+			}
+		}
+		$this->db->sql_freeresult($result);
+
+		$this->db->sql_transaction('commit');
+
+		// Log activity
+		foreach ($profile_fields as $field_ident)
+		{
+			add_log('admin', 'LOG_PROFILE_FIELD_REMOVED', $field_ident);
+		}
```

Comment:
`			$this->log->add('admin', $this->user->data['user_id'], (empty($this->user->ip)) ? '' : $this->user->ip, 'LOG_PROFILE_FIELD_REMOVED', time(), array($field_ident));`